Documentation/project-docs/AssemblyBlobs.md

```diff
@@ -0,0 +1,214 @@
+
+ - `magic`: has the value of 0x41424158 (`XABA`)
+ - `version`: a value increased every time blob format changes.
+ - `local_entry_count`: number of assemblies stored in this blob (also
+   the number of entries in the assembly descriptor table, see below)
+ - `global_entry_count`: number of entries in the index blob's (see
+   below) hash tables, all the other blobs store `0` in this field
+ - `blob_id`: a unique ID of this blob.
+ 
+## Assembly descriptor table
+
+Each blob header is followed by a table of
+`BundledAssemblyBlobHeader.local_entry_count` entries, each entry
+defined by the following structure:
+
+    struct BlobBundledAssembly
+    {
+        uint32_t data_offset;
+        uint32_t data_size;
+        uint32_t debug_data_offset;
+        uint32_t debug_data_size;
+        uint32_t config_data_offset;
+        uint32_t config_data_size;
+    };
+
+Only the `data_offset` and `data_size` fields must have a non-zero
+value, other fields describe optional data and can be set to `0`. 
+
+Individual fields have the following meanings:
+
+  - `data_offset`: offset of the assembly image data from the
+    beginning of the blob file
+  - `data_size`: number of bytes of the image data
+  - `debug_data_offset`: offset of the assembly's debug data from the
+    beginning of the blob file. A value of `0` indicates there's no
+    debug data for this assembly.
+  - `debug_data_size`: number of bytes of debug data. Can be `0` only
+    if `debug_data_offset` is `0`
+  - `config_data_offset`: offset of the assembly's config file data
+    from the  beginning of the blob file. A value of `0` indicates
+    there's no config file data for this assembly.
+  - `config_data_size`: number of bytes of config file data. Can be
+    `0` only if `config_data_offset` is `0`
+
+## Index blob
+
+Each application will contain exactly one blob with a global index -
+two tables with assembly name hashes.  All the other blobs **do not**
+contain these tables.  Two hash tables are necessary because hashes
+for 32-bit and 64-bit devices are different.
+
+The hash tables follow the [Assembly descriptor
+table](#assembly-descriptor-table) and precede the individual assembly
+streams.
+
+Placing the hash tables in a single index blob, while "wasting" a
+certain amount of memory (since 32-bit devices won't use the 64-bit
+table and vice versa), makes for simpler and faster runtime
+implementation and the amount of memory wasted isn't big (1000
+two tables which are 8kb long each, this being the amount of memory
+wasted)
+
+### Hash table format
+
+Both tables share the same format, despite the hashes themselves being
+of different sizes.  This is done to make handling of the tables
+easier on the runtime.
+
+Each entry contains, among other fields, the assembly name hash.  The
+hash value is obtained using the
+[xxHash](https://cyan4973.github.io/xxHash/) algorithm and is
+calculated **without** including the `.dll` extension.  This is done
```

Comment:
I would like an example that pertains to *satellite* assemblies: how/where is the path used?  Consider `es/libZipSharp.resources.dll` vs. `fr/libZipSharp.resources.dll`.

Reply:
The path is treated as part of the name, and we reconstruct the name with culture [here](https://github.com/xamarin/xamarin-android/pull/6311/files#diff-3157a46af8a1b1c13c7d1ad16bcfe787f40a0a07e3e7251282500e2a8af5a7f2R475-R483), using the result to calculate the hash at the runtime that matches the one calculated on the build time.